s3tests_boto3/functional/test_s3.py

```diff
@@ -2152,6 +2152,17 @@ def test_object_set_get_metadata_overwrite_to_empty():
     got = _set_get_metadata('', bucket_name)
     eq(got, '')
 
+@attr(resource='object.metadata')
+@attr(method='put')
+@attr(operation='metadata key with underscore')
+@attr(assertion='correct handling of dash and underscore in key names')
+def test_object_set_get_metadata_dash_underscore():
+    bucket_name = get_new_bucket()
+    got = _set_get_metadata('kay-fabe')
```

Comment:
since you're creating a bucket for these, you should pass `bucket_name` in the calls to `_set_get_metadata(metadata, bucket_name=None)`

Reply:
yes, of course--I was mixing them